ngraph/src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -283,20 +283,14 @@ static bool replace_transpose_with_reshape(shared_ptr<Node> transpose)
     return true;
 }
 
-static unordered_map<NodeTypeInfo, function<bool(shared_ptr<Node>)>> initialize_ops_to_simplifiers()
+bool pass::AlgebraicSimplification::run_on_function(shared_ptr<Function> f)
 {
-    return unordered_map<NodeTypeInfo, function<bool(shared_ptr<Node>)>>(
+    static const unordered_map<NodeTypeInfo, function<bool(shared_ptr<Node>)>> ops_to_simplifiers =
```

Comment:
thread sanitizer will complain that `unordered_map` is created from multiple threads in case if we run this code from parallel threads. So, better to keep as is and this `unordered_map` will be created only once during ngraph loading.

Reply:
As I know for C++ 11 it is not an issue. See https://stackoverflow.com/questions/1661529/is-meyers-implementation-of-the-singleton-pattern-thread-safe. Leaving this in global scope leads to segfault due to unfortunate order of initialization for the globals.

Reply:
good to know. But strange that on practice thread sanitizer still complains, but may be it's a false positive..

Reply:
I don't know exactly why it complains (if it does -- I didn't check), but this likely gives an answer: https://stackoverflow.com/questions/42062557/c-multithreading-is-initialization-of-a-local-static-lambda-thread-safe